Sources/_CryptoExtras/MLDSA/MLDSA_boring.swift

```diff
@@ -0,0 +1,305 @@
+        /// 
+        /// - Throws: `CryptoKitError.incorrectKeySize` if the seed is not 32 bytes long.
+        public init(seed: some DataProtocol) throws {
+            self.backing = try Backing(seed: seed)
+        }
+
+        /// The seed from which this private key was generated.
+        public var seed: Data {
+            self.backing.seed
+        }
+
+        /// The public key associated with this private key.
+        public var publicKey: PublicKey {
+            self.backing.publicKey
+        }
+
+        /// Generate a signature for the given data.
+        /// 
+        /// - Parameters: 
+        ///   - data: The message to sign.
+        ///   - context: The context to use for the signature.
+        /// 
+        /// - Returns: The signature of the message.
+        public func signature<D: DataProtocol>(for data: D, context: D? = nil) throws -> Signature {
+            try self.backing.signature(for: data, context: context)
+        }
+
+        /// The size of the private key in bytes.
+        static let bytesCount = Backing.bytesCount
+
+        fileprivate final class Backing {
+            var key: MLDSA65_private_key
+            var seed: Data
+
+            /// Initialize a ML-DSA-65 private key from a random seed.
+            init() throws {
+                self.key = .init()
+                self.seed = Data() // We have to initialize all members before `self` is captured by the closure
+
+                self.seed = try withUnsafeTemporaryAllocation(of: UInt8.self, capacity: MLDSA.seedSizeInBytes) { seedPtr in
+                    try withUnsafeTemporaryAllocation(of: UInt8.self, capacity: MLDSA.PublicKey.Backing.bytesCount) { publicKeyPtr in
+                        guard CCryptoBoringSSL_MLDSA65_generate_key(
+                            publicKeyPtr.baseAddress,
+                            seedPtr.baseAddress,
+                            &self.key
+                        ) == 1 else {
+                            throw CryptoKitError.internalBoringSSLError()
+                        }
+
+                        return Data(bytes: seedPtr.baseAddress!, count: MLDSA.seedSizeInBytes)
+                    }
+                }
+            }
+
+            /// Initialize a ML-DSA-65 private key from a seed.
+            /// 
+            /// - Parameter seed: The seed to use to generate the private key.
+            /// 
+            /// - Throws: `CryptoKitError.incorrectKeySize` if the seed is not 32 bytes long.
+            init(seed: some DataProtocol) throws {
+                guard seed.count == MLDSA.seedSizeInBytes else {
+                    throw CryptoKitError.incorrectKeySize
+                }
+
+                self.key = .init()
+                self.seed = Data(seed)
+
+                guard CCryptoBoringSSL_MLDSA65_private_key_from_seed(
+                    &self.key,
+                    Array(seed.prefix(MLDSA.seedSizeInBytes)),
```

Comment:
Let's not create an array here, nor should we use `prefix`. We know the seed size is right, and we can use `self.seed.withUnsafeBytes` to get the interior pointer there.